docs/source/tutorial/multi_node_multi_gpu_tutorial.rst

```diff
@@ -0,0 +1,252 @@
+Multi-Node-Multi-GPU GNN Training
+==================================
+
+Before doing this tutorial we recommend going through <insert single-node-multi-gpu tutorial> as a warm up.
+Our first step is to understand the basic structure of a multi-node-multi-gpu example.
+
+
+.. code-block:: python
+
+    import argparse
+    import time
+    import warnings
+
+    import torch
+    import torch.distributed as dist
+    from torch_geometric.datasets import FakeDataset
+
+    from torch_geometric.nn.models import GCN
+
+    warnings.filterwarnings("ignore")
+
+
+    _LOCAL_PROCESS_GROUP = None
+
+
+    def create_local_process_group(num_workers_per_node):
+        ...
+
+
+    def get_local_process_group():
+        ...
+
+
+    def run(device, data, world_size, model, epochs, batch_size, fan_out,
+                  split_idx, num_classes):
+        ...
```

Comment:
We don't need these func definitions right?

Reply:
these are definitely needed functions to set up the process groups for multinode

Reply:
I was asking because I see the definitions right below this section.

Reply:
oh i see what you mean. ill leave it up to @rusty1s to decide, i chose to show these as a skeleton to give an idea of the key base functions needed for multi-node but i can understand your reasoning as well.